src/services/services.ts

```diff
@@ -3345,6 +3347,20 @@ export function createLanguageService(
         );
     }
 
+    function getImports(fileName: string): readonly string[] {
+        synchronizeHostData();
+        const file = getValidSourceFile(fileName);
+        const imports: Set<string> = new Set();
+        let specifiers = file.imports;
+        specifiers = specifiers.slice(specifiers.findIndex(nodeIsSynthesized) + 1);
+        if (specifiers.length === 0) return [];
+        for (const specifier of specifiers) {
+            const name = program.getResolvedModuleFromModuleSpecifier(specifier, file)?.resolvedModule?.resolvedFileName;
+            if (name && !imports.has(name)) imports.add(name);

```

Comment:
```suggestion
if (name) imports.add(name);
```
checking for presence and then adding is not required ?